source/InjectiveTypes/WeakFactorizationSystem.lagda

```diff
@@ -0,0 +1,183 @@
+Tom de Jong, 22 May 2025
+
+An anonymous reviewer of our TYPES abstract [1] suggested that some of our
+results could be generalized to weak factorization systems. Here we consider a
+factorization system based on embeddings and maps whose fibers are all injective
+types. We are also thinking about the connections to *algebraic* weak
+factorization systems.
+
+[1] Tom de Jong and Martı́n Hötzel Escardó.
+    "Examples and counter-examples of injective types in univalent mathematics".
+    Abstract for the 31st International Conference on Types for Proofs and
+    Programs (TYPES). 2025.
+    url: https://msp.cis.strath.ac.uk/types2025/abstracts/TYPES2025_paper6.pdf.
+
+\begin{code}
+
+{-# OPTIONS --safe --without-K #-}
+
+open import UF.FunExt
+
+module InjectiveTypes.WeakFactorizationSystem
+        (fe : FunExt)
+       where
+
+open import InjectiveTypes.Blackboard fe
+
+open import MLTT.Spartan
+
+open import UF.Base
+open import UF.Embeddings
+open import UF.Equiv
+open import UF.EquivalenceExamples
+open import UF.IdEmbedding
+open import UF.Univalence
+
+\end{code}
+
+We define a fiberwise algebraically injective map to be one whose fibers are all
+algebraically injective types.
+
+NB: It may be that fiberwise flabbiness is more convenient to work with.
+
+\begin{code}
+
+fiberwise-ainjective : {A : 𝓤 ̇  } {B : 𝓥 ̇  } → (A → B) → (𝓦 𝓣 : Universe)
+                     → ((𝓦 ⊔ 𝓣) ⁺ ⊔ 𝓤 ⊔ 𝓥) ̇
+fiberwise-ainjective f 𝓦 𝓣 =
+ each-fiber-of f (λ F → ainjective-type F 𝓦 𝓣)
+
+\end{code}
+
+Any map can be factored as an embedding followed by a fiberwise algebraically
+injective map.
+
+\begin{code}
+
+embedding-fiberwise-ainjective-factorization
+ : {A : 𝓤 ̇  } {B : 𝓥 ̇  }
+ → is-univalent (𝓤 ⊔ 𝓥)
+ → (f : A → B)
+ → Σ X ꞉ (𝓤 ⊔ 𝓥) ⁺ ̇  ,
+   Σ l ꞉ (A → X) ,
+   Σ r ꞉ (X → B) , (f ＝ r ∘ l)
+                 × is-embedding l
+                 × fiberwise-ainjective r (𝓤 ⊔ 𝓥) (𝓤 ⊔ 𝓥)
+
+private
+ module factorization-construction
+         {A : 𝓤 ̇  } {B : 𝓥 ̇  } (f : A → B)
+        where
+
+  ι : (X : 𝓤' ̇  ) → X → (X → 𝓤' ̇  )
```

Comment:
You seem to have two spaces between the superscript do and the closing bracket. One is enough (for the browsers to show this properly). I notice this in other files you wrote (some of which I fixed in the past).

Reply:
Sorry, old habit.